### DISCUSSION TITLE
Question about DoDragDrop method from on Mac (IPlatformDragSource)

### DISCUSSION THREAD

**Discussion by Tenjim** [AvaloniaUI/Avalonia#15933] [Q&A] [answered]
Hi,

I'm currently on version 11.0.10. Only on MacOs, I have a weird behaviour with command modifiers to copy multiple items with Drag And Drop. It won't copy because the effects pass to DragDrop.DoDragDrop method are none in my case but this code work on Windows.

I investigate this issue and I'm seeing that IPlatformDragSource is obsolete and not in the master.
[Obsolete("This is a part of unstable API and can be changed in minor releases. Consider replacing it with alternatives or reach out developers on GitHub.")]
Do you have any idea if this behavior has been fixed in the master ?

**Comment by Tenjim** [comment]
I had updated [my repos ](https://github.com/Tenjim/DndSample) to reproduce my issue with DragAndDrop Copy effects on macos. 

When I hold Command key and drop on the third textblock from my sample, It doesn't do anything. It's a behavior specific to macos ?

**Reply by thevortexcloud** [reply]
I can't speak specifically for Mac, but I know Drag and Drop is not actually implemented on all platforms. Linux for example does not work at all. It's one of the few APIs that has some fairly major platform shaped holes in it.

**Reply by Tenjim** [reply]
This behavior on mac with command key / copy effect is there since 10.19. So yeah, I tought it was the current state of DnD on macos

**Reply by jamsoft** [reply]
Yeah, I can't get DnD working on macOS at all. Seems it only really works on Windows at the moment.

**Reply by maxkatz6** [reply]
@jamsoft  DnD is implemented on macOS, but your issue is likely related to sandboxing specifically.
<img width="1032" alt="Screenshot 2024-07-08 at 5 11 43 PM" src="https://github.com/AvaloniaUI/Avalonia/assets/3163374/d652b18c-9988-4d14-aed0-b955ae28729a">

**Reply by maxkatz6** [reply, marked as answer]
@Tenjim I am not sure if this behavior specific to macOS, Fiddler seems to ignore command key. If it's the case, can you open an issue?

**Reply by Tenjim** [reply]
> @Tenjim I am not sure if this behavior specific to macOS, Fiddler seems to ignore command key. If it's the case, can you open an issue?

Yes, I'll do it because it's a bug for me.
